testing/tests/DevExpress.ui.widgets/popup.tests.js

```diff
@@ -2735,35 +2731,51 @@ QUnit.module('positioning', {
             const visualPosition = this.getPosition();
             const expectedPosition = $target.position();
 
-            assert.deepEqual(visualPosition, expectedPosition, 'position is correct');
+            assert.deepEqual(visualPosition, expectedPosition);
         });
 
-        QUnit.test('popup should not restore position after fullScreen disable', function(assert) {
-            const visualPositionBeforeFullScreen = this.getPosition();
+        QUnit.test('popup should render on top-center position after change fullScreen option to false', function(assert) {
+            this.init({ position: 'top' });
+
+            const expectedPosition = this.getPosition();
+
+            this.reinit({
+                position: 'top',
+                fullScreen: true,
+            });
 
-            this.popup.option('fullScreen', true);
-            this.popup.option('position', { of: '#container' });
             this.popup.option('fullScreen', false);
 
-            const visualPositionAfterFullScreen = this.getPosition();
-            assert.deepEqual(visualPositionAfterFullScreen, visualPositionBeforeFullScreen, 'visual position was not changed');
+            const visualPosition = this.getPosition();
+            assert.deepEqual(visualPosition, expectedPosition);
         });
 
-        QUnit.test('popup should not restore position on rerender after fullScreen changed to false', function(assert) {
-            const visualPositionBeforeFullScreen = this.getPosition();
+        QUnit.test('popup should render on top-center position after on/off fullScreen option', function(assert) {
+            this.init({
+                position: 'top',
+            });
+
+            const initialPosition = this.getPosition();
 
             this.popup.option('fullScreen', true);
-            this.popup.option('position', { of: '#container' });
             this.popup.option('fullScreen', false);
 
-            this.popup.option('height', 'auto');
-            const visualPositionAfterFullScreen = this.getPosition();
+            const visualPosition = this.getPosition();
+            assert.deepEqual(visualPosition, initialPosition);
+        });
+
+        QUnit.test('popup should render on top-left position after change fullScreen option to true', function(assert) {
```

Comment:
It's not clear top left of what. It can be related to container / visualContainer / position.of and so on
```suggestion
        QUnit.test('popup should render at the left top corner of the window after change fullScreen option to true', function(assert) {
```